code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/ApiAuthenticator.java

```diff
@@ -55,11 +55,15 @@ public void handle(ApiRequest request) throws IOException {
         }
 
         Account authenticatedAsAccount = getAccount(request);
-        if (authenticatedAsAccount == null) {
-            throwUnauthorizated();
+        if (authenticatedAsAccount == null ||
+                !StringUtils.equals(CommonStatesConstants.ACTIVE, authenticatedAsAccount.getState())) {
+            throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
         }
 
         Set<Identity> identities = getIdentities(authenticatedAsAccount);
+        if (identities == null || identities.size() == 0) {
```

Comment:
Nit: looks like getIdentities can never return null, only an empty list, so not null check can be omitted here.